### ISSUE TITLE
Ran into problem following OSX instructions

### ISSUE THREAD

**Issue by timoffex** [emacs-lsp/lsp-haskell#67]
I followed the instructions under the "Hie Installation (OSX)" heading and got the following error:

```
Timas-MacBook-Air:Developer timoffex$ git clone https://github.com/haskell/haskell-ide-engine
Cloning into 'haskell-ide-engine'...
remote: Enumerating objects: 107, done.
remote: Counting objects: 100% (107/107), done.
remote: Compressing objects: 100% (62/62), done.
remote: Total 28008 (delta 51), reused 72 (delta 36), pack-reused 27901
Receiving objects: 100% (28008/28008), 6.66 MiB | 974.00 KiB/s, done.
Resolving deltas: 100% (15112/15112), done.
Timas-MacBook-Air:Developer timoffex$ cd haskell-ide-engine
Timas-MacBook-Air:haskell-ide-engine timoffex$ ./install.hs build-all
Error when running Shake build system:
  at want, called at src/Development/Shake/Internal/Args.hs:83:69 in shake-0.18.5-44KSA7uQF2VObxzEvLYZx:Development.Shake.Internal.Args
* Depends on: build-all
  at error, called at src/Development/Shake/Internal/Rules/File.hs:179:58 in shake-0.18.5-44KSA7uQF2VObxzEvLYZx:Development.Shake.Internal.Rules.File
* Raised the exception:
Error, file does not exist and no rule available:
  build-all
```

I've never used Shake (at least not directly). What am I missing here? Is there a typo in the instructions or do the instructions need to be amended?

Thanks!

**Comment by timoffex** [comment]
Update: I went to https://github.com/haskell/haskell-ide-engine and followed their installation from source requirements.

```
stack ./install.hs hie
```
(warning: this takes a long time and installs for the latest GHC; see end of this comment)

I ran into an Xcode-related problem which was easy to fix:

```
xcode-select: error: tool ‘xcodebuild’ requires Xcode, but active developer directory ‘/Library/Developer/CommandLineTools’ is a command line tools instance
```

Solution: see https://medium.com/@vschroeder/xcode-select-error-tool-xcodebuild-requires-xcode-but-active-developer-directory-fb7d3408c80b


After I tried `stack ./install.hs hie` again, everything built! It took about 1.5 hours, so it would have really sucked if it ended with an error (which I experienced when trying to build some other Haskell environment-related thing using Nix about a year ago, except the build time was 8+ hours).

After all of this, I realized that my existing `stack` project used an older version of GHC, and the above command builds stuff for the latest version. This caused an error in the *Messages* buffer when I opened one of my Haskell files in Emacs. The error was somewhat unhelpful: it just said it couldn't build or find "ghc-prim-<some version>". I don't remember what it was exactly. I quit Emacs, updated my resolver to the latest one, rebuilt my project (ran into #41), and reopened a Haskell file.

No errors this time, though "Flymake" in the status bar is stuck on [Wait] and most commands I've tried (jump to definition) are timing out.


Leaving issue open because the instructions in the README led to confusion. I don't understand why the instructions led to a weird error, so I don't know enough to fix them myself.

**Comment by timoffex** [comment]
Update: I got it to work!

- Flymake was stuck on [Wait] because haskell-ide-engine didn't succeed.
- "Flymake" was in the status bar because lsp-mode was selecting flymake. Setting lsp-prefer-flymake to nil (as in #41) didn't solve the problem because I needed to install flycheck-haskell (https://github.com/flycheck/flycheck-haskell).
- To get haskell-ide-engine to work in my stack project, I also had to add an "hie.yaml" file with just these contents:

```
cradle:
  stack:
```

While trying to debug this, it was helpful to look at the *hie::stderr* buffer.

Still leaving issue open in case README instructions could be improved/updated, at least for the part about installing `haskell-ide-engine`.
